pllug-presentation-system/presentation-data/domdocumentdivider.cpp

```diff
@@ -0,0 +1,125 @@
+#include "domdocumentdivider.h"
+
+const int cMaxNumOfElements = 5;
+
+std::unique_ptr<Presentation> DomDocumentDivider::import(const QByteArray &inputText)
+{
+    QByteArray htmlText = "<root>" + inputText + "</root>";
+    QDomDocument htmlDocument;
+    htmlDocument.setContent(htmlText);
+
+    QVector<QDomDocument> separatedParts = divideBySeparators(htmlDocument);
+    QVector<Slide> undividedSlides = divideByHeaders(separatedParts);
+
+    return std::make_unique<Presentation>(*divideSlides(undividedSlides));
```

Comment:
We need to change CONFIG+=c++11 to CONFIG+=c++14 in "pllug-presentation-system.pro"
 to use "std::make_unique" correctly, as this feature was implemented in C++14 standard. Please, see [std::make_unique](http://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) on cppreference.com.